Hover events added

mrityunjaygr8 opened the pull request:
mouseenter and mouseleave event listeners added so that additional data/effects can be shown on hover of the country.
Readme updated to document the same functionality

maguayo commented on src/Map.vue:
```diff
@@ -188,3 +188,21 @@
+      p.addEventListener('mouseleave', () => {
+        vm.$emit('mouseleave', this);
+      });
+    });
```
I think "path" is a better variable name than "p" IMO

frossigneux commented:
Hello @mrityunjaygr8, thanks for your PR.

Could you add an example in the doc on how to get the current selected country and its value?
Currently I get a VueComponent with many nested properties, and I don't know how to retrieve data about the current country, in order to display a tooltip.

I think also that the color bar should be deactivable with a property.

mrityunjaygr8 commented:
Hi @frossigneux 
Yes the color bar should be deactivatable via a property, I will push a patch for that soon.
Regarding the example, I'll try to add an example.

mrityunjaygr8 commented:
Hi @frossigneux 

I've updated the package now.
The colorBar is now optional using the showColorBar property.
Further, I've corrected my mistake due to which on_mouseleave and on_mouseneter were throwing extra data about the VueComponent. Now they are returning the Country's  ISO 3166 Code, which we use in the countryData propery, so we can get the value from the countryData propery itself.

In order to show the country's full name, currently I use another Object which has the ISO 3166 code as the key and Country's full name as the value.
eg. 
country_full_name = {
    "US":  "United States",
    ....
}

mulhoon commented on src/App.vue:
```diff
@@ -1,6 +1,7 @@
 <template>
 <div class="vue-world-map">
-  <Map />
+  <div :v-if="showColorBar" id="color_bar"></div>
```
```suggestion
  <div v-if="showColorBar" id="color_bar"></div>
```
Mistake here `:v-if` instead of `v-if` causes showColorBar to not work

mrityunjaygr8 replied:
good catch man,
thanks for the heads up
